lib/mayaUsd/fileio/shaderReaderRegistry.h

```diff
@@ -0,0 +1,137 @@
+//
+// Copyright 2016 Pixar
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef PXRUSDMAYA_SHADER_READER_REGISTRY_H
+#define PXRUSDMAYA_SHADER_READER_REGISTRY_H
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/fileio/primReaderArgs.h>
+#include <mayaUsd/fileio/primReaderContext.h>
+#include <mayaUsd/fileio/shaderReader.h>
+
+#include <pxr/pxr.h>
+#include <pxr/usd/sdf/path.h>
+
+#include <maya/MFnDependencyNode.h>
+
+#include <functional>
+#include <string>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+/// \class UsdMayaShaderReaderRegistry
+/// \brief Provides functionality to register and lookup USD shader reader
+/// plugins for Maya nodes.
+///
+/// Use PXRUSDMAYA_DEFINE_SHADER_READER(usdInfoId, args, ctx) to define a new
+/// reader function, or use
+/// PXRUSDMAYA_REGISTER_SHADER_READER(usdInfoId, readerClass) to register a
+/// reader class with the registry.
+///
+/// In order for the core system to discover the plugin, you need a
+/// \c plugInfo.json that contains the usdInfoId and the Maya plugin to load:
+/// \code
+/// {
+///     "UsdMaya": {
+///         "ShaderReader": {
+///             "mayaPlugin": "myMayaPlugin",
```

Comment:
The preview surface was moved to mayaUsd plugin, so it is in memory at this point in time.

This also means that we can remove https://github.com/Autodesk/maya-usd/blob/dev/lib/usd/translators/plugInfo.json without any ill effect since mayaUsd_Translator is always in memory as well.